### PR TITLE
update files

### PR DESCRIPTION
**Pull request by Spencer-Pleno**
Update yml for ansible install of droid

STILL TODO:
-Properly activate a conda environment
-Include dictionary of multiple conda environments to update

### CODE REVIEW

File: main.yml [hunk 1]
```diff
@@ -9,3 +9,9 @@
   roles:
     - plantuml
     - ntp_server
+
+- name: Setup droid user
+  hosts: localhost
+  roles:
+    - py3.12_linux_perf
```
Comment (bo-at-pleno) [inline main.yml:16]: what is this role?

File: roles/pleno_droid/defaults/main.yml [hunk 2]
```diff
@@ -1,6 +1,5 @@
 ---
-pleno_droid_rootdir: "{{ ansible_env.HOME }}/Projects/performance-study"
-pleno_droid_droid2_version: 2.0.1
-pleno_droid_droid2_venv: "{{ pleno_droid_rootdir }}/{{ pleno_droid_droid2_version }}-py3.12.3-venv"
-pleno_droid_droid1_version: 1.24.1
-pleno_droid_droid1_venv: "{{ pleno_droid_rootdir }}/{{ pleno_droid_droid1_version }}-py3.12.3-venv"
+pleno_droid_python_path: ".pyenv/versions/3.12.3/bin/python"
+pleno_droid_experimental:
+  version: 2.1.2
+  venv_path: "spencerdev311"
```
Comment (bo-at-pleno) [inline roles/pleno_droid/defaults/main.yml:5]: ?? do not compute.

File: roles/pleno_droid/tasks/main.yml [hunk 3]
```diff
@@ -1,47 +1,26 @@
...
-    path: "{{ pleno_droid_rootdir }}"
-    state: directory
-    mode: "0755"
-
-- name: (Droid2) Setup execution environment
-  ansible.builtin.pip:
-    name: pip>=23.2.1
+- name: Ensure user john doe exists
+  ansible.builtin.user:
+    name: johndoe
```
Comment (bo-at-pleno) [inline roles/pleno_droid/tasks/main.yml:4]: ?? just a placeholder?

**Review by bo-at-pleno — APPROVED** [review]
Just minor questions but LGTM afaik

File: main.yml (hunk 1 above)
Comment (antony-gigapixel-imaging) [reply main.yml:16]: This role ensures Python 3.12 exists on the `dev` server. Spencer and I will remove this requirement when we support WPL machine (i.e. `*.33`) in the future.

File: roles/pleno_droid/tasks/main.yml (hunk 3 above)
Comment (antony-gigapixel-imaging) [reply roles/pleno_droid/tasks/main.yml:4]: I don't know whether the WPL server enforces per-user security (e.g. user `wpl` restricted to `/srv/wpl` to host all venvs). So, for the sake of practicing ANsible with Spencer, we use a placeholder user account.

File: roles/pleno_droid/tasks/main.yml [hunk 4]
```diff
@@ -1,47 +1,26 @@
...
-    virtualenv: "{{ pleno_droid_droid2_venv }}"
-    extra_args: "--extra-index-url https://pypi.plenoinc.com/pypi/simple/"
-
-- name: (Droid1) Setup execution environment
+- name: (Droid) Setup execution environment
   ansible.builtin.pip:
     name: pip>=23.2.1
     state: present
-    virtualenv: "{{ pleno_droid_droid1_venv }}"
+    virtualenv: "{{ droid_user.home }}/{{ pleno_droid_experimental.venv_path }}"
```
[the remark this answers is not in the document]
Comment (antony-gigapixel-imaging) [reply roles/pleno_droid/tasks/main.yml:13]: Yes in the next PR. Spencer and I are comparing a `dict` yaml structure, or an `array` yaml structure to represent various configs (experimental/stable/latest). Comparing the pros and cons.

File: roles/pleno_droid/tasks/main.yml [hunk 5]
```diff
@@ -1,47 +1,26 @@
...
     virtualenv_site_packages: true
-    virtualenv_command: "{{ pyenv_user.home }}/.pyenv/versions/3.12.3/bin/python -m venv"
+    virtualenv_command: "{{ pyenv_user.home }}/{{ pleno_droid_python_path }} -m venv"
+  become_user: "{{ droid_user.name }}"
+  become: true
 
-- name: (Droid1) Install droid and dependencies
+- name: (Droid) Install droid and dependencies
   ansible.builtin.pip:
-    name: "pleno_droid=={{ pleno_droid_droid1_version }}"
```
Comment (antony-gigapixel-imaging) [inline roles/pleno_droid/tasks/main.yml:33]: Note to self: @Spencer-Pleno wants to use conda to manage python environments, not `venv`. Similarly, use conda to install droid, not `pip`.

Conda is not officially supported by Ansible core. But, there is a user plugin. Not sure if it has long term support.
https://github.com/Pleno-Inc/pleno-ansible/pull/15/commits/44d975151603edc3dd148b79fc783e119ea34bb2

cc'ed @bo-at-pleno  for visibility.

**Review by antony-gigapixel-imaging — APPROVED** [review]

**Pull request merged**